pkg/asset/installconfig/gcp/client.go

```diff
@@ -155,11 +158,31 @@ func (c *Client) GetPublicDNSZone(ctx context.Context, project, baseDomain strin
 		return nil, errors.Wrap(err, "failed to list DNS Zones")
 	}
 	if res == nil {
-		return nil, errors.New("no matching public DNS Zone found")
+		zoneType := "public"
+		if !publicVisibility {
+			zoneType = "private"
+		}
+		return nil, fmt.Errorf("no matching %s DNS Zone found", zoneType)
 	}
 	return res, nil
 }
 
+// GetPrivateDNSZone returns a private DNS zone for a basedomain.
+func (c *Client) GetPrivateDNSZone(ctx context.Context, project, baseDomain string) (*dns.ManagedZone, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 1*time.Minute)
+	defer cancel()
+
+	return c.GetDNSZoneWithVisibility(ctx, project, baseDomain, false)
+}
```

Comment:
This function isn't being used and can be removed from the code. As an extension, the `GetDNSZoneWithVisibility` function is also no longer required. As a result, all of this file can be reverted back into the original.